datastream-common/src/main/java/com/linkedin/datastream/metrics/BrooklinMetrics.java

```diff
@@ -5,19 +5,11 @@
  */
 package com.linkedin.datastream.metrics;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.atomic.AtomicInteger;
-
-
 /**
  * Base class for metric categories with support for metric deregistration.
- * It uses reference counting for deregistering aggregate metrics which can
- * only happen when all keyed metrics of the same name have been deregistered.
+ * Underlying DynamicMetricsManager will take care of maintaining ref-counts.
  */
 public abstract class BrooklinMetrics {
```

Comment:
We have an empty class just to deregister aggregate metrics now. Is it possible to explicitly call  deregister Aggregates in a backward compatible way?

Reply:
As mentioned in the description, "I'm keeping the existing interfaces for backward compatibility and can be cleaned up in future if required." Since, the changes are in other components, I'm keeping the changes here. Once I will migrate that code, I will remove this method from here in the future.